Mergin/utils.py

```diff
@@ -882,3 +882,19 @@ def is_number(s):
         return False
     except TypeError:
         return False
+
+
+def has_schema_change(mp, layer):
```

Comment:
proper way is to add "has_schema_change" to (py)geodiff, since at the end of the day geodiff is one that needs to decide if it is for conflict or not... @wonder-sk do you think it is worth adding it now? (I think you can combine few API calls to get the info already, but it is not easiest)

Reply:
You can do something like this:
```
geodiff.schema('sqlite', '', '/home/martin/project/file.gpkg', '/tmp/schema_local.json')
geodiff.schema('sqlite', '', '/home/martin/project/.mergin/file.gpkg', '/tmp/schema_base.json')
# now compare the two json files
```

I agree that it would be nice to have API in geodiff to check for that directly - just raised https://github.com/lutraconsulting/geodiff/issues/164 for that - but it's probably not necessary to add it right now...

Reply:
Also agreed that we should use geodiff, because there are other bits that `layer.fields()` does not cover - e.g. geometry type change, or new table added etc

Reply:
I think it is faster to add to geodiff. Making 2 tmp files and then deleting them(?), doing diff sounds like more work than implementation in geodiff for all platforms we support.